### PR TITLE
feat: update fill logic

### PR DESCRIPTION
**Pull request by anna-carroll**
supersedes #53

### CODE REVIEW

File: src/Orders.sol [hunk 1]
```diff
@@ -22,36 +22,37 @@ struct Output {
...
+    /// @param outputs - The Outputs to be transferred.
+    /// @dev NOTE that here, Output.chainId denotes the *origin* chainId.
+    /// @custom:emits Filled
+    function fill(Output[] memory outputs) external payable {
+        // transfer outputs
+        uint256 value = msg.value;
+        for (uint256 i; i < outputs.length; i++) {
+            if (outputs[i].token == address(0)) {
+                // this line should underflow if there's an attempt to spend more ETH than is attached to the transaction
+                value -= outputs[i].amount;
```
Comment (prestwich) [inline src/Orders.sol:48]: opt: future versions could crawl the list and collapse transfers to the same account, instead of having 1 transfer per output

**Review by prestwich — CHANGES_REQUESTED** [review]
just a comment change

File: src/Orders.sol [hunk 2]
```diff
@@ -22,36 +22,37 @@ struct Output {
...
+    /// @notice Emitted when Order Outputs are sent to their recipients.
+    /// @param outputs - The Outputs transferred to their recipients
+    /// @dev NOTE that here, Output.chainId denotes the *origin* chainId.
+    event Filled(Output[] outputs);
+
+    /// @notice Send the Output(s) of any number of Orders.
+    ///         The user calls `initiate` on a rollup; the Builder calls `fill` on the target chain aggregating Outputs.
+    /// @param outputs - The Outputs to be transferred.
+    /// @dev NOTE that here, Output.chainId denotes the *origin* chainId.
+    /// @custom:emits Filled
```
[the remark this answers is not in the document]
Comment (anna-carroll) [reply src/Orders.sol:41]: https://github.com/init4tech/zenith/pull/55/commits/5a874740a9d93a8b8d0f09882479410f37a6d476

File: src/Orders.sol (hunk 1 above)
Comment (anna-carroll) [reply src/Orders.sol:48]: yeah, the Solidity for this is a bit gnarly though :) the Builder should aggregate the Outputs offchain to save on gas.

Comment (prestwich) [reply src/Orders.sol:48]: builder can't aggregate outputs with different chain ids tho

**Pull request merged**